tiny_dnn/core/kernels/fully_connected_op_avx.h

```diff
@@ -32,28 +32,25 @@ namespace tiny_dnn {
 namespace kernels {
 
 inline void
-fully_connected_op_avx(const tensor_t& in_data,
-                       const vec_t&    W,
-                       const vec_t&    bias,
-                       tensor_t&       out_data,
-                       const fully_params& params,
-                       const bool      layer_parallelize) {
+fully_connected_op_avx(const Tensor<float_t, 2>& in_data,
```

Comment:
Three parameters seems ok. We can rearrange them `<kDim, kConst, type, allocator>``, so you won't have to put `float_t ` every time